src/worker.cpp

```diff
@@ -486,11 +486,32 @@ void Worker::setDeviceTreeAndJson()
 
     if (fitConfigVal.find(devTreeFromJson) != std::string::npos)
     { // fitconfig is updated and correct JSON is set.
-
         if (isSystemVPDOnDBus())
         {
-            // TODO
-            //  Restore system VPD logic should initiate from here.
+            m_backupRestoreObjPtr = new BackupRestore(m_parsedJson);
+            if (auto l_sysVpd = std::get_if<types::IPZVpdMap>(&parsedVpdMap))
+            {
+                const std::string& l_sysImKwdValue = getIMValue(*l_sysVpd);
+
+                /* System's list, for which backup and restore should be called
+                 * before publishing VPD on DBus.*/
+                const std::vector<std::string> l_systemTypes = {
```

Comment:
Why the check is after instantiating the class?

Reply:
only including BackupRestore API's implementation in this PR.